include/aws/s3/private/s3_request.h

```diff
@@ -113,6 +113,15 @@ struct aws_s3_request {
     /* Linked list node used for queuing. */
     struct aws_linked_list_node node;
 
+    /* Linked list node used for tracking the request is active from HTTP level. */
+    struct aws_linked_list_node on_going_http_node;
```

Comment:
_on_going_http_node is confusing. Maybe make it more descriptive. 
Something like `ongoing_http_request_list_node`